app/models/user.rb

```diff
@@ -3,4 +3,16 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  has_many :orders, dependent: :destroy
+  has_many :courses, through: :orders
+
+  def own_course?(course)
+    user_courses = approved_courses
+    user_courses.include? course
+  end
+
+  def approved_courses
+    orders.where(status: 'approved').map(&:course)
```

Comment:
Como order has_many :courses through, você pode fazer uma query assim pra evitar carregar no Rails e com Ruby transformar podendo gerar uma query de N+1

```suggestion
    orders.where(status: 'approved').courses
```

Reply:
Realmente não consegui o retorno de `orders.where(status: 'approved')` está sendo um array. Não se é por estar acessando direto de orders e order é `belongs_to :course` ... não sei ... vou fazer fazer algumas tentaticas aqui 
![image](https://user-images.githubusercontent.com/41846128/116432691-2b5e8780-a81f-11eb-8f7d-aa1bae73e7ce.png)

Reply:
e se fazer o contrario? será que dá?

```
courses.where(orders: { status: :approved })
```

Reply:
Deu certo assim o/